feat: gateway support for EN

ly0va opened the pull request:
## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

ly0va committed:
It compiles

StanislavBreadless committed:
cargo fix

StanislavBreadless committed:
resolve lint

StanislavBreadless committed:
latest contracts

StanislavBreadless committed:
upd configs

StanislavBreadless committed:
Update contracts (#2288)

## What ❔

Brings updates from this PR:
https://github.com/matter-labs/era-contracts/pull/541

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.

StanislavBreadless committed:
upd contracts

StanislavBreadless committed:
upd contracts

StanislavBreadless committed:
return from accidental change

StanislavBreadless committed:
sync with base

StanislavBreadless committed:
upd contracts

StanislavBreadless committed:
Merge pull request #2295 from matter-labs/sb-custom-da-with-sl

feat!: Custom DA with SyncLayer

StanislavBreadless committed:
wip contracts

StanislavBreadless committed:
upd contracts

StanislavBreadless committed:
basic integration

StanislavBreadless committed:
make withdrawals on top of one chain pass

StanislavBreadless committed:
some basic api for proofs of batch inclusion

ly0va committed:
Add priorityTree layer

ly0va committed:
Bump contracts

StanislavBreadless committed:
agg root in db

ly0va committed:
Add the forgotten file

ly0va committed:
Merge branch 'sync-layer-stable' into lyova-integrate-priority-merkle-tree_

StanislavBreadless committed:
withdrawals work for 1 layer and agg proof is corect

ly0va committed:
Bump contracts

ly0va committed:
zk fmt

kelemeno committed:
feat: more system contracts support, WIP (#2273)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.

---------

Co-authored-by: Raid Ateir <ra@matterlabs.dev>
Co-authored-by: Raid5594 <52794079+Raid5594@users.noreply.github.com>
Co-authored-by: Stanislav Breadless <stanislavbezkor@gmail.com>

StanislavBreadless committed:
comms work

ly0va committed:
Fix caching bug

ly0va committed:
Tests and fixes

StanislavBreadless committed:
new contracts

StanislavBreadless committed:
sync with base

StanislavBreadless committed:
int tests with withdrawals work

kelemeno committed:
feat: blobhash-compilation-fix-support (#2336)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.

StanislavBreadless committed:
partially resolve unit tests

StanislavBreadless committed:
upd env

kelemeno committed:
feat: bump contracts (#2342)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

ly0va committed:
Fix PQ->PT switch

ly0va committed:
Bump contracts

ly0va committed:
Merge remote-tracking branch 'origin/sync-layer-stable' into lyova-integrate-priority-merkle-tree_

ly0va committed:
Fix sqlx data

ly0va committed:
Fix config

ly0va committed:
Temporarily do deposits sequentially

kelemeno committed:
Merge branch 'sync-layer-stable' of github.com:matter-labs/zksync-era into sb-l3-l1-comms-integration

kelemeno committed:
bump contracts. small config fix

kelemeno committed:
contracts.toml

ly0va committed:
Bump contracts

ly0va committed:
Fix lint

ly0va committed:
Update config

ly0va committed:
Merge pull request #2335 from matter-labs/lyova-integrate-priority-merkle-tree_

feat: integrate priority tree

ly0va committed:
Point contracts to sync-layer-stable

koloz193 committed:
feat(pubdata): send call encoded pubdata (#2360)

## What ❔

Update server to encode pubdata with parameters for DA validator call.

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

ly0va committed:
Merge remote-tracking branch 'origin/main' into sync-layer-stable

ly0va committed:
zk fmt

ly0va committed:
Merge remote-tracking branch 'origin/sync-layer-stable' into lyova-merged-sync-layer-stable

ly0va committed:
Fix Cargo.lock

ly0va committed:
Fix workflow file

ly0va committed:
Bump contracts

ly0va committed:
Fix deprecated warning

ly0va committed:
Config fixes

koloz193 committed:
Merge branch 'sync-layer-stable' into sb-l3-l1-comms-integration

koloz193 committed:
add message root init to tests

koloz193 committed:
bump sleep term

koloz193 committed:
fix execute batch txns not being generated

koloz193 committed:
toml update

koloz193 committed:
update contracts

koloz193 committed:
bump contracts

koloz193 committed:
bump sleep before finalize

koloz193 committed:
increase timeout

koloz193 committed:
revert changes to tests

koloz193 committed:
change waitFinalize for waitUntilBlockFinalized

koloz193 committed:
bump balances

koloz193 committed:
Revert "bump balances"

This reverts commit 03a598bf89d9f04a1b281f7ba5e19724ffc3f9c2.

koloz193 committed:
skip tests

StanislavBreadless committed:
lint fix + remove describe only

StanislavBreadless committed:
cargo deny fix

StanislavBreadless committed:
try increase the timeout

StanislavBreadless committed:
increase timeout for erc20

StanislavBreadless committed:
make timeout even higher

StanislavBreadless committed:
larger capacity for base docker pubdata

ly0va committed:
Fix some todos

StanislavBreadless committed:
remove fixme comments

StanislavBreadless committed:
Merge pull request #2339 from matter-labs/sb-l3-l1-comms-integration

feat: L3 L1 comms integration

ly0va committed:
Merge remote-tracking branch 'origin/sync-layer-stable' into lyova-merged-sync-layer-stable

ly0va committed:
Fix stuff

StanislavBreadless committed:
basic test fix

StanislavBreadless committed:
remove unneeded method

StanislavBreadless committed:
fix server launch

StanislavBreadless committed:
add some comments

StanislavBreadless committed:
fixme comment

ly0va committed:
Fix prover tests

ly0va committed:
Update lockfile

ly0va committed:
Fix prover tests again

ly0va committed:
Update Cargo.lock

ly0va committed:
Update deny.toml

ly0va committed:
Fix compilation

StanislavBreadless committed:
add tokio runtime

StanislavBreadless committed:
fix how stdout works

ly0va committed:
Merge remote-tracking branch 'origin/main' into lyova-merged-sync-layer-stable

ly0va committed:
Resolve post-merge errors

ly0va committed:
Fix EN tests

ly0va committed:
Bump contracts

ly0va committed:
Update env

StanislavBreadless committed:
fix init

StanislavBreadless committed:
fmt

StanislavBreadless committed:
fix more int tests

StanislavBreadless committed:
fmt

StanislavBreadless committed:
fmt

StanislavBreadless committed:
sync with upstream

StanislavBreadless committed:
upd contracts

StanislavBreadless committed:
int tests passing

StanislavBreadless committed:
turn off unused test

ly0va committed:
Merge pull request #2504 from matter-labs/sb-fix-rust-unit-tests

fix: Fix rust unit tests

ly0va committed:
Fix lints

ly0va committed:
Fix Cargo.lock

ly0va committed:
Bump contracts

ly0va committed:
Lints

ly0va committed:
Bump contracts

ly0va committed:
Merge pull request #2395 from matter-labs/lyova-merged-sync-layer-stable

feat(sync-layer): megre main & resolve todos

StanislavBreadless committed:
compiles at least partially

StanislavBreadless committed:
fix config

StanislavBreadless committed:
lint

StanislavBreadless committed:
comment out upgrade test also

StanislavBreadless committed:
Merge pull request #2597 from matter-labs/sb-merge-main-sync-layer-stable

chore: Merge main into sync layer stable

StanislavBreadless committed:
resolve env hack

StanislavBreadless committed:
small refactor

StanislavBreadless committed:
fix lint

StanislavBreadless committed:
lint fix

StanislavBreadless committed:
Merge pull request #2619 from matter-labs/sb-resolve-env-hack

chore: Resolve env hack

ly0va committed:
Preliminary impl

ly0va committed:
Compiles

ly0va committed:
Rename

ly0va committed:
Configs

ly0va committed:
Merge branch 'sync-layer-stable' into lyova-en-gateway-support

perekopskiy commented on core/node/api_server/src/web3/state.rs:
```diff
@@ -106,6 +106,7 @@ pub struct InternalApiConfig {
     pub state_transition_proxy_addr: Option<Address>,
     pub transparent_proxy_admin_addr: Option<Address>,
     pub user_facing_diamond_proxy_addr: Address,
+    pub diamond_proxy_addr: Address,
```
Can you please explain what's the difference between `user_facing_diamond_proxy_addr` and `diamond_proxy_addr`?

perekopskiy commented on core/node/consistency_checker/src/lib.rs:
```diff
@@ -511,14 +541,25 @@ impl ConsistencyChecker {
-        tracing::debug!("Performing sanity checks for diamond proxy contract {address:?}");
+        }
+
+        let (address, client) = regular.or(migration).unwrap();
 
+        tracing::debug!("Performing sanity checks for diamond proxy contract {address:?}");
```
nit: let's check both contracts if they are present

perekopskiy commented on core/node/node_framework/src/implementations/layers/consistency_checker.rs:
```diff
@@ -75,6 +87,17 @@ impl WiringLayer for ConsistencyCheckerLayer {
+            consistency_checker = consistency_checker.with_migration_setup(
+                input
+                    .migration_client
+                    .expect("Migration client not provided")
```
nit: return error instead of panicking

perekopskiy commented on core/node/node_framework/src/implementations/layers/tree_data_fetcher.rs:
```diff
@@ -61,17 +75,28 @@ impl WiringLayer for TreeDataFetcherLayer {
+            fetcher = fetcher.with_migration_setup(
+                input
+                    .migration_client
+                    .expect("Migration client not configured")
```
same

perekopskiy commented on core/node/consistency_checker/src/lib.rs:
```diff
@@ -312,13 +312,21 @@ pub struct ConsistencyChecker {
     max_batches_to_recheck: u32,
     sleep_interval: Duration,
     l1_client: Box<DynClient<L1>>,
+    migration_setup: Option<MigrationSetup>,
```
nit: let's include "gateway" word in struct or var name so it's clear it's migration to gateway

ly0va replied:
It's not necessarily to gateway. Even though for now we don't support it, we can be migrating to other settlement layers (Gateway->L1, or sideways)

perekopskiy commented on core/lib/basic_types/src/protocol_version.rs:
```diff
@@ -174,6 +174,10 @@ impl ProtocolVersionId {
         self >= &ProtocolVersionId::Version23
     }
+
+    pub fn is_post_sync_layer(&self) -> bool {
+        self >= &ProtocolVersionId::Version25
+    }
```
it's not needed for now right? I don't mind keeping it if you think it will be 100% needed later

StanislavBreadless committed:
more substantial sync layer -> gateway rename

StanislavBreadless committed:
using latest contracts

StanislavBreadless committed:
use the new contracts

StanislavBreadless committed:
Merge pull request #2650 from matter-labs/sb-sync-kl-sync-layer-reorg

chore: Use contracts from kl sync layer reorg

StanislavBreadless committed:
sync with main

StanislavBreadless committed:
rust unit tests pass

StanislavBreadless committed:
complete migration

StanislavBreadless committed:
zk

StanislavBreadless committed:
merge main

StanislavBreadless committed:
sync with main for the second time

StanislavBreadless committed:
fix lint

StanislavBreadless committed:
Merge pull request #2664 from matter-labs/sb-merge-main-into-sync-layer-stable-2

chore: Merge main into sync layer stable 2

ly0va committed:
Resolve comments

ly0va committed:
Merge branch 'sync-layer-stable' into lyova-en-gateway-support

ly0va committed:
Fix compilation

ly0va committed:
Fix zk toolbox

StanislavBreadless committed:
int tests work

StanislavBreadless committed:
fmt

StanislavBreadless committed:
fix config

StanislavBreadless committed:
upd contracts scripts

StanislavBreadless committed:
updated contracts commit

StanislavBreadless committed:
int tests pass

StanislavBreadless committed:
reduce owner usage in init

StanislavBreadless committed:
fmt

StanislavBreadless committed:
use latest sync layer stable

StanislavBreadless committed:
remove unneeded change

StanislavBreadless committed:
surpress error

StanislavBreadless committed:
use latest contracts

StanislavBreadless committed:
another way to fetch l2 token addr

StanislavBreadless committed:
new contracts

StanislavBreadless committed:
latest commit

StanislavBreadless committed:
sync with base

StanislavBreadless committed:
Merge pull request #2677 from matter-labs/sb-sync-kl-reorg

chore: Contracts from kl/sync-layer-reorg

StanislavBreadless committed:
latest contracts

StanislavBreadless committed:
use sync layer stable

StanislavBreadless committed:
fix gas limit in sdk

StanislavBreadless committed:
Merge pull request #2681 from matter-labs/sb-make-owner-init-footprint-smaller

feat: make owner init footprint smaller

kelemeno committed:
fix: script fixes (#2701)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

tomg10 committed:
merge changes from master

Signed-off-by: tomg10 <lemures64@gmail.com>

tomg10 committed:
experiment

Signed-off-by: tomg10 <lemures64@gmail.com>

tomg10 committed:
scripts

Signed-off-by: tomg10 <lemures64@gmail.com>

StanislavBreadless committed:
fix setting env

StanislavBreadless committed:
Merge pull request #2712 from matter-labs/sb-new-env-for-sl-stable

chore: Fix setting env

ly0va committed:
Merge remote-tracking branch 'origin/sync-layer-stable' into sync-layer-stable-tjg

ly0va committed:
zk fmt

ly0va committed:
Fix imports

ly0va committed:
Try fix CI

ly0va committed:
Try fix CI

tomg10 committed:
bugfix

Signed-off-by: tomg10 <lemures64@gmail.com>

tomg10 committed:
bugfix

Signed-off-by: tomg10 <lemures64@gmail.com>

tomg10 committed:
remove debug

Signed-off-by: tomg10 <lemures64@gmail.com>

tomg10 committed:
bugfix

Signed-off-by: tomg10 <lemures64@gmail.com>

tomg10 committed:
linter fix

Signed-off-by: tomg10 <lemures64@gmail.com>

ly0va committed:
Merge pull request #2710 from matter-labs/sync-layer-stable-tjg

feat: merge main into sync-layer-stable

ly0va committed:
Merge remote-tracking branch 'origin/sync-layer-stable' into lyova-en-gateway-support

koloz193 committed:
feat(contracts: integrate protocol defense changes

koloz193 committed:
Merge branch 'main' into zk-protocol-defense

StanislavBreadless committed:
chore: Port kl sync layer reorg to the stable branch (#2747)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

koloz193 committed:
update protocol version and multivm bootloaders

koloz193 committed:
Merge branch 'main' into zk-protocol-defense

koloz193 committed:
bump contracts

koloz193 committed:
bump contracts

koloz193 committed:
Merge branch 'main' into zk-protocol-defense

ly0va committed:
Fetch chain_ids

StanislavBreadless committed:
add submodules ignored folder

StanislavBreadless committed:
Merge branch 'main' into zk-protocol-defense

StanislavBreadless committed:
use new contracts

StanislavBreadless committed:
sync with zach branch

StanislavBreadless committed:
unit tests pass

StanislavBreadless committed:
upd config

StanislavBreadless committed:
lint fix

StanislavBreadless committed:
Merge pull request #2762 from matter-labs/sb-merge-contracts-dev-into-sl-stable

chore: Merge contracts dev into sl stable

kelemeno committed:
chore: message-root-support (#2706)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

---------

Co-authored-by: Stanislav Breadless <stanislavbezkor@gmail.com>

ly0va committed:
Compiled

ly0va committed:
Fix lints and unit tests

tomg10 committed:
feat(eth-watch): redesign to support Gateway (#2775)

Signed-off-by: tomg10 <lemures64@gmail.com>

StanislavBreadless committed:
feat(contracts): Remove priority op info from gw (#2813)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

ly0va committed:
It sort of works

StanislavBreadless committed:
basic sync with main

StanislavBreadless committed:
return commented out tests

StanislavBreadless committed:
fix lint

StanislavBreadless committed:
upd contracts

StanislavBreadless committed:
resolve one more conflict

StanislavBreadless committed:
hopefully fix rust unit test

StanislavBreadless committed:
hopefully fix docker init

StanislavBreadless committed:
maybe fix unit tests

StanislavBreadless committed:
wait a bit longer

StanislavBreadless committed:
sync with main

StanislavBreadless committed:
fix lock

StanislavBreadless committed:
turn off loadtest

StanislavBreadless committed:
fix lock

ly0va committed:
Add compute address script

StanislavBreadless committed:
Merge pull request #2858 from matter-labs/sb-sync-main-sl-stable

chore: Sync gateway with main

ly0va committed:
Update config

ly0va committed:
Merge branch 'sync-layer-stable' into lyova-en-gateway-support

ly0va committed:
It works now

kelemeno committed:
feat: Kl split-asset-router-support (#2790)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

ly0va committed:
Try to restore EN CI

ly0va committed:
Please run CI

ly0va committed:
CI fixes

ly0va committed:
CI fixes

kelemeno committed:
chore: bump contracts (#2902)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

ly0va committed:
Fix unit tests

ly0va committed:
Fix more unit tests

ly0va committed:
Typo

ly0va committed:
Move deserialization logic

StanislavBreadless committed:
feat(toolbox): Port toolbox to work with gateway (#2910)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

---------

Co-authored-by: Danil <deniallugo@gmail.com>

StanislavBreadless committed:
use synced contracts

StanislavBreadless committed:
merge with main

StanislavBreadless committed:
latest contracts

StanislavBreadless committed:
fix rust unit

StanislavBreadless committed:
comment out unused methods

StanislavBreadless committed:
lint fix

StanislavBreadless committed:
solve lint

StanislavBreadless committed:
Merge pull request #2932 from matter-labs/sb-sync-layer-toolbox-functions-with-main

chore: Merge main into toolbox functions

ly0va committed:
Merge remote-tracking branch 'origin/sync-layer-stable' into lyova-en-gateway-support

ly0va committed:
Fix config

deniallugo-ml commented:
Closing this PR as it has been open without activity for over a year and the codebase has moved on significantly since it was opened. The changes would need a substantial rebase at minimum, and the original context may no longer apply.

If this addresses something that is still relevant, please open a fresh PR against the current `main` with an updated diff — we are happy to review it.

Thank you for the contribution attempt.
